[nuklear] fatal error: 'malloc.h' file not found

**Is your feature request related to a problem? Please describe.**

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES C)

set(src src/main.c)

find_path(NUKLEAR_INCLUDE_DIRS "nuklear.h")

add_executable(${PROJECT_NAME} ${src})
target_include_directories(${PROJECT_NAME} PRIVATE ${NUKLEAR_INCLUDE_DIRS})
````

### `src/main.c`
```c
#include <nuklear.h>
#include <malloc.h>
#include <allegro5/allegro.h>
#define MAX_MEMORY 1024

int main(void) {
    /* init gui state */
    struct nk_context ctx;
    NkAllegro5Font *font;
    font = nk_allegro5_font_create_from_file("../../../extra_font/Roboto-Regular.ttf", 12, 0);
    nk_init_fixed(&ctx, calloc(1, MAX_MEMORY), MAX_MEMORY, &font);

    enum {EASY, HARD};
    static int op = EASY;
    static float value = 0.6f;
    static int i =  20;

    if (nk_begin(&ctx, "Show", nk_rect(50, 50, 220, 220),
                 NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_CLOSABLE)) {
        /* fixed widget pixel width */
        nk_layout_row_static(&ctx, 30, 80, 1);
        if (nk_button_label(&ctx, "button")) {
            /* event handling */
        }

        /* fixed widget window ratio width */
        nk_layout_row_dynamic(&ctx, 30, 2);
        if (nk_option_label(&ctx, "easy", op == EASY)) op = EASY;
        if (nk_option_label(&ctx, "hard", op == HARD)) op = HARD;

        /* custom widget pixel width */
        nk_layout_row_begin(&ctx, NK_STATIC, 30, 2);
        {
            nk_layout_row_push(&ctx, 50);
            nk_label(&ctx, "Volume:", NK_TEXT_LEFT);
            nk_layout_row_push(&ctx, 110);
            nk_slider_float(&ctx, 0, &value, 1.0f, 0.1f);
        }
        nk_layout_row_end(&ctx);
    }
    nk_end(&ctx);
}
```

### `shell`
```sh
$ [path]/vcpkg install nuklear
$ mkdir build && cd $_
$ cmake -DCMAKE_TOOLCHAIN_FILE=~/repos/vcpkg/scripts/buildsystems/vcpkg.cmake ..
$ cmake --build .
/tmp/b/src/main.c:2:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/pp.dir/src/main.c.o] Error 1
make[1]: *** [CMakeFiles/pp.dir/all] Error 2
make: *** [all] Error 2
```

**Proposed solution**
Would be nice if the example would build

**Describe alternatives you've considered**
```c
#if defined(macintosh) || defined(Macintosh) || defined(__APPLE__) && defined(__MACH__)
#include <stdlib.h>
#else
#include <malloc.h>
#endif
````
(but would need to be more careful, there is a list of macros on sourceforge lined up with where to find malloc, so I bunch of these checks are required)

But it could find neither `#include "nuklear.h"` nor `#include <nuklear.h>`…

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES C)

set(src src/main.c)

find_path(NUKLEAR_INCLUDE_DIRS "nuklear.h")
include(FindPackageHandleStandardArgs)
include(SelectLibraryConfigurations)
find_path(ALLEGRO_INCLUDES NAMES allegro.h HINTS ${CURRENT_INSTALLED_DIR} PATH_SUFFIXES allegro5)
find_library(ALLEGRO_LIBRARY_RELEASE NAMES allegro-static PATH_SUFFIXES lib PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}" NO_DEFAULT_PATH REQUIRED)
find_library(ALLEGRO_LIBRARY_DEBUG NAMES allegro-debug-static PATH_SUFFIXES lib PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug" NO_DEFAULT_PATH REQUIRED)
select_library_configurations(ALLEGRO)

add_executable(${PROJECT_NAME} ${src})
target_include_directories(${PROJECT_NAME} PRIVATE ${ALLEGRO_INCLUDES} ${NUKLEAR_INCLUDE_DIRS})
target_link_libraries(${PROJECT_NAME} PRIVATE ${ALLEGRO_LIBRARY_DEBUG})
```

### `src/main.c`
```c
#include <nuklear.h>
#include <malloc.h>
#include <allegro5/allegro.h>
#define MAX_MEMORY 1024

int main(void) {
    /* init gui state */
    struct nk_context ctx;
    NkAllegro5Font *font;
    font = nk_allegro5_font_create_from_file("../../../extra_font/Roboto-Regular.ttf", 12, 0);
    nk_init_fixed(&ctx, calloc(1, MAX_MEMORY), MAX_MEMORY, &font);

    enum {EASY, HARD};
    static int op = EASY;
    static float value = 0.6f;
    static int i =  20;

    if (nk_begin(&ctx, "Show", nk_rect(50, 50, 220, 220),
                 NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_CLOSABLE)) {
        /* fixed widget pixel width */
        nk_layout_row_static(&ctx, 30, 80, 1);
        if (nk_button_label(&ctx, "button")) {
            /* event handling */
        }

        /* fixed widget window ratio width */
        nk_layout_row_dynamic(&ctx, 30, 2);
        if (nk_option_label(&ctx, "easy", op == EASY)) op = EASY;
        if (nk_option_label(&ctx, "hard", op == HARD)) op = HARD;

        /* custom widget pixel width */
        nk_layout_row_begin(&ctx, NK_STATIC, 30, 2);
        {
            nk_layout_row_push(&ctx, 50);
            nk_label(&ctx, "Volume:", NK_TEXT_LEFT);
            nk_layout_row_push(&ctx, 110);
            nk_slider_float(&ctx, 0, &value, 1.0f, 0.1f);
        }
        nk_layout_row_end(&ctx);
    }
    nk_end(&ctx);
}
```

### `sh`
```sh
$ /tmp/vcpkg/vcpkg install allegro5 nuklear
$ rm -rf build && mkdir $_ && cd $_
/tmp/nuklear/build $ cmake -DCMAKE_TOOLCHAIN_FILE=/tmp/vcpkg/scripts/buildsystems/vcpkg.cmake ..
-- The C compiler identification is AppleClang 12.0.0.12000032
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/nuklear/build
/tmp/nuklear/build $ cmake --build .
[ 50%] Building C object CMakeFiles/pp.dir/src/main.c.o
/tmp/nuklear/src/main.c:10:5: error: use of undeclared identifier 'NkAllegro5Font'
    NkAllegro5Font *font;
    ^
/tmp/nuklear/src/main.c:10:21: error: use of undeclared identifier 'font'; did you mean 'for'?
    NkAllegro5Font *font;
                    ^~~~
                    for
/tmp/nuklear/src/main.c:10:21: error: expected expression
/tmp/nuklear/src/main.c:11:5: error: use of undeclared identifier 'font'
    font = nk_allegro5_font_create_from_file("../../../extra_font/Roboto-Regular.ttf", 12, 0);
    ^
/tmp/nuklear/src/main.c:11:12: error: implicit declaration of function 'nk_allegro5_font_create_from_file' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
    font = nk_allegro5_font_create_from_file("../../../extra_font/Roboto-Regular.ttf", 12, 0);
           ^
/tmp/nuklear/src/main.c:12:61: error: use of undeclared identifier 'font'
    nk_init_fixed(&ctx, calloc(1, MAX_MEMORY), MAX_MEMORY, &font);
                                                            ^
6 errors generated.
make[2]: *** [CMakeFiles/pp.dir/src/main.c.o] Error 1
make[1]: *** [CMakeFiles/pp.dir/all] Error 2
make: *** [all] Error 2
```

@JackBoosY 

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES C)

set(src src/main.c)

add_executable("${PROJECT_NAME}" "${src}")

find_package(unofficial-nuklear CONFIG REQUIRED)
target_link_libraries("${PROJECT_NAME}" PRIVATE nuklear)
```

### `src/main.c`
```c
#include <nuklear.h>
//#include <malloc.h>
#include <stdlib.h>
#include <allegro5/allegro.h>
#define MAX_MEMORY 1024

int main(void) {
    /* init gui state */
    struct nk_context ctx;
    NkAllegro5Font *font;
    font = nk_allegro5_font_create_from_file("../../../extra_font/Roboto-Regular.ttf", 12, 0);
    nk_init_fixed(&ctx, calloc(1, MAX_MEMORY), MAX_MEMORY, &font);

    enum {EASY, HARD};
    static int op = EASY;
    static float value = 0.6f;
    static int i =  20;

    if (nk_begin(&ctx, "Show", nk_rect(50, 50, 220, 220),
                 NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_CLOSABLE)) {
        /* fixed widget pixel width */
        nk_layout_row_static(&ctx, 30, 80, 1);
        if (nk_button_label(&ctx, "button")) {
            /* event handling */
        }

        /* fixed widget window ratio width */
        nk_layout_row_dynamic(&ctx, 30, 2);
        if (nk_option_label(&ctx, "easy", op == EASY)) op = EASY;
        if (nk_option_label(&ctx, "hard", op == HARD)) op = HARD;

        /* custom widget pixel width */
        nk_layout_row_begin(&ctx, NK_STATIC, 30, 2);
        {
            nk_layout_row_push(&ctx, 50);
            nk_label(&ctx, "Volume:", NK_TEXT_LEFT);
            nk_layout_row_push(&ctx, 110);
            nk_slider_float(&ctx, 0, &value, 1.0f, 0.1f);
        }
        nk_layout_row_end(&ctx);
    }
    nk_end(&ctx);
}
```

### `sh`
@ bcb1c2a0
```
$ git clone https://github.com/microsoft/vcpkg && cd vcpkg
$ gh pr checkout 18881
$ ./bootstrap-vcpkg.sh
$ ./vcpkg install nuklear
$ mkdir [project_dir]/build && cd "$_"
$ cmake ..
$ cmake --build .
[ 50%] Building C object CMakeFiles/pp.dir/src/main.c.o
test_repos/nuklear/src/main.c:4:10: fatal error: 'allegro5/allegro.h' file not found
#include <allegro5/allegro.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [CMakeFiles/pp.dir/build.make:76: CMakeFiles/pp.dir/src/main.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/pp.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
```

On macOS 11.4 (20F71) with Apple clang version 12.0.5 (clang-1205.0.22.11)

After a `./vcpkg install nuklear allegro5` and removing + recreating the build dir `cmake ..` gave me:
```
CMake Error at CMakeLists.txt:16 (find_library):
  Could not find ALLEGRO_LIBRARY_DEBUG using the following names:
  allegro-debug
```

PS: Your allegro dance… not sure if this is assumed knowledge for everyone using `vcpkg`?

--- a/ports/nuklear/CMakeLists.txt
+++ b/ports/nuklear/CMakeLists.txt
@@ -18,7 +18,8 @@ install(
     ARCHIVE DESTINATION lib
 )
 
-install(FILES "${CMAKE_CURRENT_LIST_DIR}/src/nuklear.h" DESTINATION "include/nuklear")
+install(FILES "${CMAKE_CURRENT_LIST_DIR}/nuklear.h" DESTINATION "include/nuklear")
+
 
 install(EXPORT unofficial-nuklear FILE unofficial-nuklear-config.cmake DESTINATION share/unofficial-nuklear)
 

--- a/ports/nuklear/CMakeLists.txt
+++ b/ports/nuklear/CMakeLists.txt
@@ -1,0 +1,31 @@
+cmake_minimum_required(VERSION 3.16)
+project(Nuklear LANGUAGES C CXX)
+
+option(INSTALL_EXAMPLE "Install the example code" OFF)
+option(INSTALL_DEMO "Install the demo code" OFF)
+
+add_library(nuklear INTERFACE)
+
+target_include_directories(nuklear INTERFACE $<INSTALL_INTERFACE:include/nuklear>)
+target_compile_definitions(nuklear INTERFACE NK_IMPLEMENTATION)
+
+# Installation
+install(
+    TARGETS nuklear
+    EXPORT unofficial-nuklear
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
+
+install(FILES "${CMAKE_CURRENT_LIST_DIR}/src/nuklear.h" DESTINATION "include/nuklear")
+
+install(EXPORT unofficial-nuklear FILE unofficial-nuklear-config.cmake DESTINATION share/unofficial-nuklear)
+
+if (INSTALL_EXAMPLE)
+    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/example" DESTINATION share/example)
+endif()
+
+if (INSTALL_DEMO)
+    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/demo" DESTINATION share/demo)
+endif()

--- a/ports/nuklear/CMakeLists.txt
+++ b/ports/nuklear/CMakeLists.txt
@@ -27,5 +27,6 @@ if (INSTALL_EXAMPLE)
 endif()
 
 if (INSTALL_DEMO)
-    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/demo" DESTINATION share/demo)
+    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/demo" DESTINATION share/nuklear)
+
 endif()

--- a/ports/nuklear/CMakeLists.txt
+++ b/ports/nuklear/CMakeLists.txt
@@ -24,7 +24,8 @@ install(FILES "${CMAKE_CURRENT_LIST_DIR}/nuklear.h" DESTINATION "include/nuklear
 install(EXPORT unofficial-nuklear FILE unofficial-nuklear-config.cmake DESTINATION share/unofficial-nuklear)
 
 if (INSTALL_EXAMPLE)
-    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/example" DESTINATION share/example)
+    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/example" DESTINATION share/nuklear)
+
 endif()
 
 if (INSTALL_DEMO)

--- a/ports/nuklear/CONTROL
+++ b/ports/nuklear/CONTROL
@@ -1,4 +1,0 @@
-Source: nuklear
-Version: 2021-03-18
-Homepage: https://github.com/Immediate-Mode-UI/Nuklear
-Description: This is a minimal state immediate mode graphical user interface toolkit written in ANSI C and licensed under public domain

--- a/ports/nuklear/portfile.cmake
+++ b/ports/nuklear/portfile.cmake
@@ -16,7 +16,6 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 
 vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
     OPTIONS ${FEATURE_OPTIONS}
 )
 

--- a/ports/nuklear/portfile.cmake
+++ b/ports/nuklear/portfile.cmake
@@ -5,6 +5,26 @@ vcpkg_from_github(
     SHA512 ce064dff721111749d4056717879f42d3e24bb94655dd2b04c137eb7391d2c90d0b1b95155912c100b537f74fd150aedc48e0ac85eb72963c66e35ac81048323
     HEAD_REF master
 )
-file(INSTALL ${SOURCE_PATH}/nuklear.h DESTINATION ${CURRENT_PACKAGES_DIR}/include)
-file(INSTALL ${SOURCE_PATH}/Readme.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/nuklear RENAME copyright)
+
+file(COPY "${CURRENT_PORT_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        example INSTALL_EXAMPLE
+        demo    INSTALL_DEMO
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS ${FEATURE_OPTIONS}
+)
+
+vcpkg_cmake_install()
 vcpkg_copy_pdbs()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-nuklear)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+
+file(INSTALL "${SOURCE_PATH}/Readme.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/nuklear/vcpkg.json
+++ b/ports/nuklear/vcpkg.json
@@ -1,0 +1,25 @@
+{
+  "name": "nuklear",
+  "version-date": "2021-03-18",
+  "port-version": 1,
+  "description": "This is a minimal state immediate mode graphical user interface toolkit written in ANSI C and licensed under public domain",
+  "homepage": "https://github.com/Immediate-Mode-UI/Nuklear",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
+  "features": {
+    "demo": {
+      "description": "Install demo files"
+    },
+    "example": {
+      "description": "Install example files"
+    }
+  }
+}

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4390,7 +4390,7 @@
     },
     "nuklear": {
       "baseline": "2021-03-18",
-      "port-version": 0
+      "port-version": 1
     },
     "numactl": {
       "baseline": "2.0.12",

--- a/versions/n-/nuklear.json
+++ b/versions/n-/nuklear.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "868bfa99f7aeda30fbb2cab19fe6e81962858274",
+      "git-tree": "9c243e51fbb0d9fa4562a3e4a8eba9a7fb2e0cdd",
       "version-date": "2021-03-18",
       "port-version": 1
     },

--- a/versions/n-/nuklear.json
+++ b/versions/n-/nuklear.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f68acd72a4e21fa3c9f3395e579790eede7a382d",
+      "version-date": "2021-03-18",
+      "port-version": 1
+    },
+    {
       "git-tree": "2a8abfe81ed0757650f10d7fd0e2ef8ecd484ae6",
       "version-string": "2021-03-18",
       "port-version": 0

--- a/versions/n-/nuklear.json
+++ b/versions/n-/nuklear.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "f68acd72a4e21fa3c9f3395e579790eede7a382d",
+      "git-tree": "868bfa99f7aeda30fbb2cab19fe6e81962858274",
       "version-date": "2021-03-18",
       "port-version": 1
     },

